Neos.Flow/Classes/Mvc/Routing/Dto/ResolveContext.php

```diff
@@ -62,6 +63,14 @@ public function __construct(UriInterface $baseUri, array $routeValues, bool $for
         $this->routeValues = $routeValues;
         $this->forceAbsoluteUri = $forceAbsoluteUri;
         $this->uriPathPrefix = $uriPathPrefix;
+
+        // Only add base uri path for absolute uri, in case of relative uri the uri has to be relative to the given base uri
+        if ($forceAbsoluteUri && $baseUri->getPath()) {
+            $this->uriPathPrefix =  Files::getUnixStylePath(Files::concatenatePaths([$baseUri->getPath(), $uriPathPrefix])) . '/';
+        }
+        if ($forceAbsoluteUri) {
+            $this->uriPathPrefix = '/' . ltrim($this->uriPathPrefix, '/');
+        }
```

Comment:
You could simplify the two conditions into sth like
```php
if ($forceAbsoluteUri) {
    $this->uriPathPrefix = '/' . ltrim($this->uriPathPrefix, '/');
    if ($baseUri->getPath() !== '') {
        $this->uriPathPrefix = rtrim($baseUri->getPath(), '\/') . $this->uriPathPrefix;
    }
}
```
(untested)

Reply:
Done. Works as expected